Sources/Plasma/PubUtilLib/plNetGameLib/Private/plNglAuth.cpp

```diff
@@ -1327,8 +1326,7 @@ static CliAuConn * GetConnIncRef (const char tag[]) {
 }
 
 //============================================================================
-static void UnlinkAndAbandonConn_CS (CliAuConn * conn) {
-    s_conns.Unlink(conn);
+static void AbandonConn(CliAuConn* conn) {
```

Comment:
I think this is my last question, but, just to verify... Does the caller not need to have the critical section locked anymore? I'm assuming *no* because this is no longer accessing any static variables, and you dropped `_CS` from the function name. If *yes*, we probably want to retain the `_CS` suffix on the function name.

Reply:
Exactly. As far as I can tell, the `s_critsect` lock was only needed to safely update `s_conns`, so now that this function doesn't do that anymore, it should be safe to call without the `s_critsect` locked (although currently all calls still have the lock).

That said, now that I'm re-reading this... maybe `AbandonConn` should lock on `conn->critsect` instead, because it manipulates a few fields directly instead of just calling methods? Honestly, I'm not quite sure which parts need which lock exactly (and I think the existing code is a bit inconsistent too).

Reply:
It's all a little muddy, for sure, so I took a look at the code. It looks like `CliAuConn`'s ping timer reconnect stuff what requires the lock on `CliAuConn::critsect`, so I don't think we need to lock that here. OTOH, the code *does* seem to lock `s_critsect` when accessing or mutating any `CliAuConn` instance's `cancelId` field (see `NotifyConnSocketConnectFailed` and the other notification callbacks), so I think we still need to lock `s_critsect` before calling this function and rename it to `AbandonConn_CS`. I'm assuming the other CliConn variants have the same ownership model :/

Reply:
> Alternatively, you could just lock `s_critsect` inside `AbandonConn`

That seems like the best solution, yeah.